python/mmSolver/tools/removesolvernodes/tool.py

```diff
@@ -0,0 +1,110 @@
+# Copyright (C) 2020 John Smith.
+#
+# This file is part of mmSolver.
+#
+# mmSolver is free software: you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# mmSolver is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with mmSolver.  If not, see <https://www.gnu.org/licenses/>.
+#
+"""
+Remove Solver Nodes - user facing.
+"""
+from functools import wraps
+
+import mmSolver.logger
+import mmSolver.api as mmapi
+import maya.cmds as cmds
+import mmSolver._api.utils as api_utils
+
+LOG = mmSolver.logger.get_logger()
+
+def undoStack(func):
+    @wraps(func)
+    def wrap(*args, **kwargs):
+        cmds.undoInfo(openChunk=True)
+        try:
+            return func(*args, **kwargs)
+        except Exception:
+            raise
+        finally:
+            cmds.undoInfo(closeChunk=True)
+    return wrap
+
+def filter_nodes(what_to_delete_dict):
+    nodes = cmds.ls(long=True) or []
+    cmds.select([])
+    node_categories = mmapi.filter_nodes_into_categories(nodes)
+    unknown_node_found = False
+    print 'filter_nodes run'
+    all_list_to_delete = list()
+    for key in what_to_delete_dict.keys():
```

Comment:
This loop is not needed. You can equally write:
```python
if what_to_delete_dict.get('markers') is not None:
    pass  # Do some marker stuff
elif what_to_delete_dict.get('bundles') is not None:
    pass  # Do some bundle stuff
```
This also removes one level of indentation, which I always like to avoid if possible.

Additionally, I like to avoid the line continuation character `\`, if possible.
```python
all_list_to_delete += \
    collect_nodes(node_categories, mode='marker')
```
I think this...
```python
all_list_to_delete += collect_nodes(
    node_categories, mode='marker')
```
is more readable and "pretty".